.github/twitter.py

```diff
@@ -0,0 +1,50 @@
+import tweepy
+import requests
+import xml.etree.ElementTree as ET
+import os
+import sys
+
+def main():
+    client = tweepy.Client(
+        consumer_key=os.getenv("API_KEY"),
+        consumer_secret=os.getenv("SECRET_KEY"),
+        access_token=os.getenv("ACCESS_TOKEN"),
+        access_token_secret=os.getenv("ACCESS_SECRET")
+    )
+
+    previous_atom = requests.get("https://squ1rrel.dev/atom.xml", allow_redirects=True).text
+    prev_links_to_items = {i.find("link").text.strip(): i for i in ET.fromstring(previous_atom)[0].findall('item')}
+    current_links_to_items = {i.find("link").text.strip(): i for i in ET.parse('_site/atom.xml').getroot()[0].findall('item')}
+
+    new_articles = { k : current_links_to_items[k] for k in set(current_links_to_items) - set(prev_links_to_items) }
+
+    if len(new_articles) > 0:
+        if len(new_articles) == 1:
+            try:
+                link = list(new_articles.keys())[0]
+                author = list(new_articles.values())[0].find("author").find("name").text
+                tag = list(new_articles.values())[0].findall("category")[1].attrib['term'].replace(" ", "")
+                ctf = list(new_articles.values())[0].findall("category")[0].attrib['term'].replace(" ", "")
+                client.create_tweet(text=f'New #{tag} writeup from {author}! #{ctf}\nhttps://squ1rrel.dev{link}')
+            except AttributeError as inst:
+                    print("Field missing: ", inst)
+                    sys.exit(1)
+            except Exception as inst:
+                print(f"Tweet for {list(new_articles.keys())[0]} not posted: {inst}")
+        else:
+            tweet_text = "New writeup!"
+            for i in list(new_articles.keys()):
+                try:
```

Comment:
I think you could remove the try/catch and just let it throw an Exception on any failure? that way the action will fail, we'll get an email about it, and we can fix it right away no matter what the problem is bc we probably won't see print statements

(here and the other try/catch above)